diesel/src/pg/metadata_lookup.rs

```diff
@@ -1,40 +1,43 @@
 #![allow(unused_parens)] // FIXME: Remove this attribute once false positive is resolved.
 
 use super::backend::{FailedToLookupTypeError, InnerPgTypeMetadata};
-use super::{PgConnection, PgTypeMetadata};
+use super::{Pg, PgTypeMetadata};
 use crate::prelude::*;
 
 use std::borrow::Cow;
 use std::cell::RefCell;
 use std::collections::HashMap;
 
 /// Determines the OID of types at runtime
-#[allow(missing_debug_implementations)]
-#[repr(transparent)]
-pub struct PgMetadataLookup {
-    conn: PgConnection,
-}
-
-impl PgMetadataLookup {
-    #[allow(clippy::new_ret_no_self)]
-    pub(crate) fn new(conn: &PgConnection) -> &Self {
-        unsafe { &*(conn as *const PgConnection as *const PgMetadataLookup) }
-    }
-
+///
+/// Most connections do not have to implement this manually.
+/// It is much easier to implement [`GetPgMetadataCache`] for a type that implements `Connection<Backend=Pg>`.
+/// This will cause this trait to be auto implemented.
+///
+/// If this is implemented manually, make sure to use some sort of cache so that lookup
+/// don't cause a database query for types that have been seen before.
```

Comment:
I would change the wording here to something like this, as no custom connection should actual implement this trait manually:
```suggestion
/// Custom implementations of `Connection<Backend = Pg>` should not implement this trait directly. 
/// Instead [`GetPgMetadataCache`] should be implemented, afterwards the generic implementation will provide 
/// the necessary functions to perform the type lookup.
```